src/course-unit/course-xblock/CourseXBlock.jsx

```diff
@@ -64,13 +68,17 @@ const CourseXBlock = ({
 
   useEffect(() => {
     // if this item has been newly added, scroll to it.
-    if (courseXBlockElementRef.current && shouldScroll) {
+    if (courseXBlockElementRef.current && (shouldScroll || isScrolledToElement)) {
       scrollToElement(courseXBlockElementRef.current);
     }
-  }, []);
+  }, [isScrolledToElement]);
```

Comment:
FYI, there is another way to do this without using react state at all. 

- Give the element a unique ID e.g. xblock-id-abcde
- Set the url hash to the above ID, i.e. "...some/url/page#xblock-id-abcde"
- Use the ":target" pseudo-class to apply the glow style. 

If the URL fragment matches an ID the browser should automatically scroll to it, and it applies the `:target` pseudo-class [ref](https://developer.mozilla.org/en-US/docs/Web/CSS/:target). 

This means by using this mechanism you can avoid any code for scrolling, and applying a class. Also changing this value doesn't cause the page to refresh since it's client-side only.

The only disadvantage is that this is never passed to the server. What that means is that the server can't perform any action based on this value. Here since there is no server this is a moot point, but worth mentioning anyway.

Reply:
Certainly, that makes sense, and keeping this part of the JavaScript code also opens up other possibilities. No need for a refactoring already.